app/src/main/java/fi/helsinki/opintoni/integration/pagemetadata/SpringPageMetaDataHttpClient.java

```diff
@@ -31,20 +35,21 @@ public SpringPageMetaDataHttpClient(RestTemplate metaDataRestTemplate) {
     }
 
     @Override
-    public String getPageBody(String pageUrl) {
+    public Optional<String> getPageBody(String pageUrl) {
+        Optional<String> pageBody = Optional.empty();
         try {
             HttpHeaders headers = new HttpHeaders();
             headers.setAccept(Lists.newArrayList(MediaType.TEXT_HTML));
-            headers.add("User-Agent", "Mozilla");
-            HttpEntity<String> entity = new HttpEntity<>("parameters", headers);
+            headers.add(USER_AGENT_KEY, USER_AGENT);
+            HttpEntity<String> entity = new HttpEntity<>(PARAMETERS_KEY, headers);
 
             ResponseEntity<String> response = metaDataRestTemplate.exchange(pageUrl, HttpMethod.GET, entity, String.class);
-            if (!response.getStatusCode().equals(HttpStatus.OK)) {
-                throw new Exception("Meta data request for url " + pageUrl + " failed with status code " + response.getStatusCode());
+            if (response.getStatusCode().equals(HttpStatus.OK)) {
+                pageBody = Optional.ofNullable(response.getBody());
             }
-            return response.getBody();
-        } catch (Exception e) {
-            throw new BadRequestException(e.getMessage());
-        }
+        } catch (Exception e) {}
```

Comment:
Pitäisikö heittää poikkeus jos http-pyyntö feilaa, kuten aiemmin? Poikkeus mäppäytyisi siten käyttäjälle järkevänä http-responsena `GlobalExceptionHandlers`in kautta.